### ISSUE TITLE
"Type[type]" is incompatible with "Type[GenericAlias]"

### ISSUE THREAD

**Issue by Chris-fullerton** [python/typeshed#10204]
I decide to report the bug here because i saw the note [here](https://github.com/microsoft/pyright/blob/main/.github/ISSUE_TEMPLATE/bug_report.md#:~:text=Note%3A%20if%20you%20are%20reporting%20a%20wrong%20signature%20of%20a%20function%20or%20a%20class%20in%20the%20standard%20library%2C%20then%20the%20typeshed%20tracker%20is%20better%20suited%20for%20this%20report%3A%20https%3A//github.com/python/typeshed/issues.):
> Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

---

### My Env:

1. execute `python` in cmd:

  ```
Python 3.10.11 (main, Apr 20 2023, 13:58:42) [Clang 14.0.6 ] on darwin
  ```

2. VScode `Pylance` setting:
  ```
Python › Analysis: Type Checking Mode: basic
  ```

### My Code:

1. function `parameterized_isinstance` with type-hint:

  ```
def parameterized_isinstance(var: Sequence, var_type: types.GenericAlias) -> bool:
  ```

2. use `parameterized_isinstance` function in some other place:

  ```
assert parameterized_isinstance(paths, tuple[str | pathlib.Path]) is True
  ```

### The error note from `Pylance`:

```
Argument of type "Type[tuple[str | Path]]" cannot be assigned to parameter "var_type" of type "GenericAlias" in function "parameterized_isinstance"
  "Type[type]" is incompatible with "Type[GenericAlias]"Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)
(module) pathlib
```

### What have I tried:

1. execute `type(tuple[str | pathlib.Path])`: get `<class 'types.GenericAlias'>` as result

**Comment by JelleZijlstra** [comment]
This issue seems more of a pyright problem, as basic types like `GenericAlias` are generally handled directly by type checkers. (cc @erictraut).

However, I think pyright is correct here. The fact that `tuple[str | Path]` is a GenericAlias at runtime is an implementation detail, and type checkers aren't expected to recognize the exact type.

**Comment by erictraut** [comment]
In the Python type system, `tuple[str | Path]` is not a subtype of `GenericAlias`. As Jelle said, `GenericAlias` is an implementation detail. You generally should not use `GenericAlias` in a type annotation if you want type checking to work.

**Comment by Chris-fullerton** [comment]
> In the Python type system, `tuple[str | Path]` is not a subtype of `GenericAlias`. As Jelle said, `GenericAlias` is an implementation detail. You generally should not use `GenericAlias` in a type annotation if you want type checking to work.

You are right. After I removed `types.GenericAlias` from type annotation, type checking works perfectly again. 

```diff
- def parameterized_isinstance(var: Sequence, var_type: types.GenericAlias) -> bool:
+ def parameterized_isinstance(var: Sequence, var_type) -> bool:
```
